### DISCUSSION TITLE
v1.2.0

### DISCUSSION THREAD

**Discussion by rileyjmurray** [cvxpy/cvxpy#1692] [General]
### CVXPY 1.2

This release marks a big milestone in CVXPY's development. It's the first time we've incremented the minor version number since releasing CVXPY 1.1 in June 2020. Since then we've added many new features and improved CVXPY's efficiency in important ways. A summary of those changes -- including many which were released with little fanfare between CVXPY 1.1.1 and 1.1.18 -- can be found on [cvxpy.org](https://www.cvxpy.org/version/1.2/updates/index.html). Changes specific to CVXPY 1.2 include:

* Four new and improved "atoms" for use in optimization modeling: ``xexp``, ``partial_trace``, ``partial_transpose``, and ``kron``. The latter three atoms significantly expand CVXPY's modeling capabilities for matrix representations of tensor products; they'll be especially useful for quantum information applications.
* Two new interfaces to numerical solvers. CVXPY can now interface with [Google OR Tools](https://developers.google.com/optimization) to call [GLOP](https://en.wikipedia.org/wiki/GLOP) and [PDLP](https://developers.google.com/optimization/lp/pdlp_math).
* Support for Python versions 3.7 to 3.10.

We've also grown in ways that can't be seen from changes to source code alone. We've adopted open [governance principles](https://github.com/cvxpy/org/blob/main/governance.md), become a [NumFOCUS affiliated project](https://numfocus.org/sponsored-projects/affiliated-projects), and -- starting this week -- we're adopting [semantic versioning](https://semver.org/).

#### Semantic versioning

Our adoption of semantic versioning will fundamentally change the way we approach CVXPY's maintenance and development. The most observable change is that new features will only be released in major or minor releases, as opposed to patch releases. Since CVXPY receives new feature contributions on a regular basis, that means you can expect minor releases from us much more often: multiple times per year instead of once in two years. It also means we'll support multiple minor-release series at any given time. Right now we provide bugfix support for CVXPY 1.1 and 1.2. Once CVXPY 1.3 comes out later this year, we'll provide bugfix support for CVXPY 1.1, 1.2, and 1.3. 

While this approach creates more work for day-to-day maintenance, it has two major benefits:

1. It gives us space to heavily refactor CVXPY's back-end for improved efficiency in the future. This will be important for CVXPY users who want to scale their convex optimization workflows to larger and more sophisticated problems.

2. It makes it easier for us to publicly recognize and encourage CVXPY's many volunteer contributors. This is crucial for the long-term health of CVXPY as an open-source software project.

Our adoption of semantic versioning is an ongoing process. Stay tuned for announcements on our [Discord server](https://discord.gg/4urRQeGBCr), [website](https://www.cvxpy.org/), or [Twitter](https://twitter.com/cvxpy_team) for more information.

#### Who made this possible?

CVXPY 1.2.0 includes contributions from 15 people across more than 25 pull requests. In no particular order, those contributors are

* @phschiele (PRs #1607, #1617, #1624, #1628)
* @xinyueshen (PR #1660)
* @allenlawrence94 (PR #1621)
* @lochsh (PR #1677)
* @PTNobel (PR #1648)
* @Michael-git96 (PR #1684)
* @chrisyeh96 (PR #1616)
* @TimonKnigge (PRs #1637, #1646)
* @dcajasn (PR #1640)
* @parthb83 (PRs #1641, #1683)
* @mlubin (PRs #1655, #1682)
* @AtsushiSakai (PR #1665)
* @h-vetinari (for [conda-forge PR 68](https://github.com/conda-forge/cvxpy-feedstock/pull/68/commits))
* @SteveDiamond (PRs #1606, #1659, #1669)
* @rileyjmurray (PRs #1598, #1643, #1670, #1688)

Among those listed above, we would like to call special attention to @phschiele, @Michael-git96, and @dcajasn -- each of whom has made contributions to CVXPY prior to version 1.1.18. Those recurring contributions are instrumental to CVXPY's success.

On behalf of the CVXPY project maintainers,
Riley Murray
CC: @akshayka @SteveDiamond, @bstellato 

<hr /><em>This discussion was created from the release <a href='https://github.com/cvxpy/cvxpy/releases/tag/v1.2.0'>v1.2.0</a>.</em>

**Comment by SteveDiamond** [comment]
Thanks @rileyjmurray for leading the 1.2 launch!
